core/service.go

```diff
@@ -377,3 +406,36 @@ func (s *Service) ProcessTransactionMessage(msg p2p.Message) error {
 
 	return nil
 }
+
+// handle authority and randomness changes over transitions from one epoch to the next
+func (s *Service) handleConsensusDigest(header *types.Header) (err error) {
+	var item types.DigestItem
+	for _, digest := range header.Digest {
+		item, err = types.DecodeDigestItem(digest)
+		if err != nil {
+			return err
+		}
+
+		if item.Type() == types.ConsensusDigestType {
+			break
+		}
+	}
+
+	// TODO: if this block is the first in the epoch and it doesn't have a consensus digest, this is an error
```

Comment:
Why not return an error? Not sure I understand the todo item here.

Reply:
we don't have a way to check what slot in the epoch we're at from core, I can open an issue for it

Reply:
#599